test/e2e/network/ingressclass.go

```diff
@@ -65,11 +87,32 @@ var _ = common.SIGDescribe("IngressClass [Feature:Ingress]", func() {
 		framework.ExpectNoError(err)
 		defer deleteIngressClass(cs, ingressClass1.Name)
 
-		ingress, err := createBasicIngress(cs, f.Namespace.Name)
-		framework.ExpectNoError(err)
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		lastFailure := ""
+
+		// the admission controller may take a few seconds to observe the ingress classes
+		if err := wait.Poll(time.Second, time.Minute, func() (bool, error) {
```

Comment:
The downside is, of course, that if the system is actually failing, we will burn a whole minute retrying to figure that out.  Should this be shorter?

Reply:
1 minute is not bad in terms of the e2e ... I'm more worried about the interval, client-go on 50x and network errors retries "silently 10 times" before returning  , can we use something like 5 seconds?
Is not like the test will ddos but in the context of 100s of tests everything contributes to generate noise

Reply:
1 minute here, 1 minute there, but OK.  It's only on the fail path.

Reply:
the bar is too low :(  https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/e2e-tests.md

> [Slow]: If a test takes more than five minutes to run (by itself or in parallel with many other tests), it is labeled [Slow]. This partition allows us to run almost all of our tests quickly in parallel, without waiting for the stragglers to finish.

The problem is that to absorb CI infra and resources issues and avoid flakes, also thinking on consumers like conformance that run in very constrained CIs (github actions is very very constrained)  we can not be very strict ... but we should not be too flexible too ... it is hard to find the right balance